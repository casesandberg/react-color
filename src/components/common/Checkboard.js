'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

let _checkboardCache = {}

function renderCheckboard(c1: string, c2: string, size: number): any {
  if (typeof document == 'undefined') return null // Dont Render On Server
  var canvas: any = document.createElement('canvas')
  canvas.width = canvas.height = size * 2
  var ctx = canvas.getContext('2d')
  if (!ctx) return null // If no context can be found, return early.
  ctx.fillStyle = c1
  ctx.fillRect(0, 0, canvas.width, canvas.height)
  ctx.fillStyle = c2
  ctx.fillRect(0, 0, size, size)
  ctx.translate(size, size)
  ctx.fillRect(0, 0, size, size)
  return canvas.toDataURL()
}

function getCheckboard(c1: string, c2: string, size: number): any {
  var key = c1 + ',' + c2 + ',' + size

  if (_checkboardCache[key]) {
    return _checkboardCache[key]
  } else {
    var checkboard = renderCheckboard(c1, c2, size)
    _checkboardCache[key] = checkboard
    return checkboard
  }
}

export class Checkboard extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    var background = getCheckboard(this.props.white, this.props.grey, this.props.size)
    return {
      'default': {
        grid: {
          Absolute: '0px 0px 0px 0px',
          background: 'url(' + background + ') center left',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="grid" ref="grid"></div>
    )
  }

}

Checkboard.defaultProps = {
  size: 8,
  white: '#fff',
  grey: '#e6e6e6',
}

export default Checkboard
