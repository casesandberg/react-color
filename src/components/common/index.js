export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
