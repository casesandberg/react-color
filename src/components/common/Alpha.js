'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
