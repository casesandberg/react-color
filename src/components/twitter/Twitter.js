import React from 'react'
import PropTypes from 'prop-types'
import reactCSS from 'reactcss'
import map from 'lodash/map'
import color from '../../helpers/color'

import { ColorWrap, EditableInput, Swatch } from '../common'

export const Twitter = ({ onChange, onSwatchHover, hex, colors, width, triangle,
  className = '' }) => {
  const styles = reactCSS({
    'default': {
      card: {
        width,
        background: '#fff',
        border: '0 solid rgba(0,0,0,0.25)',
        boxShadow: '0 1px 4px rgba(0,0,0,0.25)',
        borderRadius: '4px',
        position: 'relative',
      },
      body: {
        padding: '15px 9px 9px 15px',
      },
      label: {
        fontSize: '18px',
        color: '#fff',
      },
      triangle: {
        width: '0px',
        height: '0px',
        borderStyle: 'solid',
        borderWidth: '0 9px 10px 9px',
        borderColor: 'transparent transparent #fff transparent',
        position: 'absolute',
      },
      triangleShadow: {
        width: '0px',
        height: '0px',
        borderStyle: 'solid',
        borderWidth: '0 9px 10px 9px',
        borderColor: 'transparent transparent rgba(0,0,0,.1) transparent',
        position: 'absolute',
      },
      hash: {
        background: '#F0F0F0',
        height: '30px',
        width: '30px',
        borderRadius: '4px 0 0 4px',
        float: 'left',
        color: '#98A1A4',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      },
      input: {
        width: '100px',
        fontSize: '14px',
        color: '#666',
        border: '0px',
        outline: 'none',
        height: '28px',
        boxShadow: 'inset 0 0 0 1px #F0F0F0',
        boxSizing: 'content-box',
        borderRadius: '0 4px 4px 0',
        float: 'left',
        paddingLeft: '8px',
      },
      swatch: {
        width: '30px',
        height: '30px',
        float: 'left',
        borderRadius: '4px',
        margin: '0 6px 6px 0',
      },
      clear: {
        clear: 'both',
      },
    },
    'hide-triangle': {
      triangle: {
        display: 'none',
      },
      triangleShadow: {
        display: 'none',
      },
    },
    'top-left-triangle': {
      triangle: {
        top: '-10px',
        left: '12px',
      },
      triangleShadow: {
        top: '-11px',
        left: '12px',
      },
    },
    'top-right-triangle': {
      triangle: {
        top: '-10px',
        right: '12px',
      },
      triangleShadow: {
        top: '-11px',
        right: '12px',
      },
    },
  }, {
    'hide-triangle': triangle === 'hide',
    'top-left-triangle': triangle === 'top-left',
    'top-right-triangle': triangle === 'top-right',
  })

  const handleChange = (hexcode, e) => {
    color.isValidHex(hexcode) && onChange({
      hex: hexcode,
      source: 'hex',
    }, e)
  }

  return (
    <div style={ styles.card } className={ `twitter-picker ${ className }` }>
      <div style={ styles.triangleShadow } />
      <div style={ styles.triangle } />

      <div style={ styles.body }>
        { map(colors, (c, i) => {
          return (
            <Swatch
              key={ i }
              color={ c }
              hex={ c }
              style={ styles.swatch }
              onClick={ handleChange }
              onHover={ onSwatchHover }
              focusStyle={{
                boxShadow: `0 0 4px ${ c }`,
              }}
            />
          )
        }) }
        <div style={ styles.hash }>#</div>
        <EditableInput
          style={{ input: styles.input }}
          value={ hex.replace('#', '') }
          onChange={ handleChange }
        />
        <div style={ styles.clear } />
      </div>
    </div>
  )
}

Twitter.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  triangle: PropTypes.oneOf(['hide', 'top-left', 'top-right']),
  colors: PropTypes.arrayOf(PropTypes.string),
}

Twitter.defaultProps = {
  width: 276,
  colors: ['#FF6900', '#FCB900', '#7BDCB5', '#00D084', '#8ED1FC', '#0693E3',
    '#ABB8C3', '#EB144C', '#F78DA7', '#9900EF'],
  triangle: 'top-left',
}

export default ColorWrap(Twitter)
