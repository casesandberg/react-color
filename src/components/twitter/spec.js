/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';
import { red } from '../../helpers/color'

import Twitter from './Twitter'
import { Swatch } from '../common'

test('Twitter renders correctly', () => {
  const tree = renderer.create(
    <Twitter { ...red } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Twitter `triangle="none"`', () => {
  const tree = renderer.create(
    <Twitter { ...red } triangle="none" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Twitter `triangle="top-right"`', () => {
  const tree = renderer.create(
    <Twitter { ...red } triangle="top-right" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})


test('Twitter with onSwatchHover events correctly', () => {
  const hoverSpy = jest.fn()
  const tree = mount(
    <Twitter onSwatchHover={hoverSpy} />
  )
  expect(hoverSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('mouseOver')

  expect(hoverSpy).toHaveBeenCalled()
})