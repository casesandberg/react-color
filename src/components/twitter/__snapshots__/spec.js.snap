exports[`test Twitter \`triangle="none"\` 1`] = `
<div
  className="twitter-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": "276px",
    }
  }>
  <div
    style={
      Object {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "position": "absolute",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "position": "absolute",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "padding": "15px 9px 9px 15px",
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FF6900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FF6900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FCB900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FCB900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#7BDCB5",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#7BDCB5" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#00D084",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#00D084" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#8ED1FC",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#8ED1FC" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#0693E3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#0693E3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#ABB8C3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#ABB8C3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#EB144C",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#EB144C" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#F78DA7",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#F78DA7" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#9900EF",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#9900EF" />
    <div
      style={
        Object {
          "MozBorderRadius": "4px 0 0 4px",
          "OBorderRadius": "4px 0 0 4px",
          "WebkitBorderRadius": "4px 0 0 4px",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "msBorderRadius": "4px 0 0 4px",
          "width": "30px",
        }
      }>
      #
    </div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }>
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="ff691f"
        style={
          Object {
            "MozBorderRadius": "0 4px 4px 0",
            "MozBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "OBorderRadius": "0 4px 4px 0",
            "OBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "WebkitBorderRadius": "0 4px 4px 0",
            "WebkitBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "msBorderRadius": "0 4px 4px 0",
            "msBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="" />
    </div>
    <div
      style={
        Object {
          "clear": "both",
        }
      } />
  </div>
</div>
`;

exports[`test Twitter \`triangle="top-right"\` 1`] = `
<div
  className="twitter-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": "276px",
    }
  }>
  <div
    style={
      Object {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "position": "absolute",
        "right": "12px",
        "top": "-11px",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "position": "absolute",
        "right": "12px",
        "top": "-10px",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "padding": "15px 9px 9px 15px",
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FF6900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FF6900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FCB900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FCB900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#7BDCB5",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#7BDCB5" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#00D084",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#00D084" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#8ED1FC",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#8ED1FC" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#0693E3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#0693E3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#ABB8C3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#ABB8C3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#EB144C",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#EB144C" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#F78DA7",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#F78DA7" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#9900EF",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#9900EF" />
    <div
      style={
        Object {
          "MozBorderRadius": "4px 0 0 4px",
          "OBorderRadius": "4px 0 0 4px",
          "WebkitBorderRadius": "4px 0 0 4px",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "msBorderRadius": "4px 0 0 4px",
          "width": "30px",
        }
      }>
      #
    </div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }>
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="ff691f"
        style={
          Object {
            "MozBorderRadius": "0 4px 4px 0",
            "MozBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "OBorderRadius": "0 4px 4px 0",
            "OBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "WebkitBorderRadius": "0 4px 4px 0",
            "WebkitBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "msBorderRadius": "0 4px 4px 0",
            "msBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="" />
    </div>
    <div
      style={
        Object {
          "clear": "both",
        }
      } />
  </div>
</div>
`;

exports[`test Twitter renders correctly 1`] = `
<div
  className="twitter-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": "276px",
    }
  }>
  <div
    style={
      Object {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-11px",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-10px",
        "width": "0px",
      }
    } />
  <div
    style={
      Object {
        "padding": "15px 9px 9px 15px",
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FF6900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FF6900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#FCB900",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#FCB900" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#7BDCB5",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#7BDCB5" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#00D084",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#00D084" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#8ED1FC",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#8ED1FC" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#0693E3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#0693E3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#ABB8C3",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#ABB8C3" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#EB144C",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#EB144C" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#F78DA7",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#F78DA7" />
    <div
      onClick={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#9900EF",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "30px",
          "margin": "0 6px 6px 0",
          "msBorderRadius": "4px",
          "width": "30px",
        }
      }
      title="#9900EF" />
    <div
      style={
        Object {
          "MozBorderRadius": "4px 0 0 4px",
          "OBorderRadius": "4px 0 0 4px",
          "WebkitBorderRadius": "4px 0 0 4px",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "msBorderRadius": "4px 0 0 4px",
          "width": "30px",
        }
      }>
      #
    </div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }>
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="ff691f"
        style={
          Object {
            "MozBorderRadius": "0 4px 4px 0",
            "MozBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "OBorderRadius": "0 4px 4px 0",
            "OBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "WebkitBorderRadius": "0 4px 4px 0",
            "WebkitBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "msBorderRadius": "0 4px 4px 0",
            "msBoxShadow": "inset 0 0 0 1px #F0F0F0",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="" />
    </div>
    <div
      style={
        Object {
          "clear": "both",
        }
      } />
  </div>
</div>
`;
